packages-next/fields/src/types/relationship/Implementation.ts

```diff
@@ -91,7 +91,12 @@ export class Relationship<P extends string> extends Implementation<P> {
       const filterArgs = refList.getGraphqlFilterFragment().join('\n');
       return [
         `${this.path}(${filterArgs}): [${refList.gqlNames.outputTypeName}!]!`,
-        this.withMeta ? `_${this.path}Meta(${filterArgs}): _QueryMeta` : '',
+        this.withMeta
+          ? `_${this.path}Meta(${filterArgs}): _QueryMeta @deprecated(reason: "This query will be removed in a future version. Please use ${this.path}Count instead.")`
+          : '',
+        this.withMeta
+          ? `${this.path}Count(${`where: ${refList.gqlNames.whereInputName}! = {}`}): Int!`
```

Comment:
This needs to be nullable (and the root one should be as well if it isn't currently)